docs/source/optim.rst

```diff
@@ -303,6 +314,185 @@ algorithms.
+ensuring existing parameters are mapped correctly, while missing ones (like the bypass layer) remain unchanged
+(as initialized in this example).
+This approach enables smooth loading and resuming of the optimizer state despite model changes.
+The new bypass layer will be trained from scratch::
+
+    class Model1(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.fc = nn.Linear(5, 5)
+
+        def forward(self, x):
+            return self.fc(x) + x
+
+
+    model = Model1()
+    optimizer = optim.SGD(model.named_parameters(), lr=0.01, momentum=0.9)
+    # training..
+    torch.save(optimizer.state_dict(), PATH)
+
+    class Model_bypass(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.fc = nn.Linear(5, 5)
+            self.bypass = nn.Linear(5, 5, bias=False)
+            torch.nn.init.eye_(self.bypass.weight)
+
+        def forward(self, x):
+            return self.fc(x) + self.bypass(x)
+
+    model2 = Model_bypass()
+    optimizer2 = optim.SGD(model2.named_parameters(), lr=0.01, momentum=0.9)
+
+    def adapt_state_dict_missing_param(optimizer, state_dict):
+        adapted_state_dict = deepcopy(optimizer.state_dict())
+        # Copy setup parameters (lr, weight_decay, etc.), in case they differ in the loaded state dict.
+        for k, v in state_dict['param_groups'][0].items():
+            if k not in ['params', 'param_names']:
+                adapted_state_dict['param_groups'][0][k] = v
+
+        lookup_dict = {
+            'fc.weight': 'fc.weight',
+            'fc.bias': 'fc.bias',
+            'bypass.weight': None,
+        }
+
+        clone_deepcopy = lambda d: {k: (v.clone() if isinstance(v, torch.Tensor) else deepcopy(v)) for k, v in d.items()}
+        for param_id, param_name in zip(
+                optimizer.state_dict()['param_groups'][0]['params'],
+                optimizer.state_dict()['param_groups'][0]['param_names']):
+            name_in_loaded = lookup_dict[param_name]
+            if name_in_loaded in state_dict['param_groups'][0]['param_names']:
+                index_in_loaded_list = state_dict['param_groups'][0]['param_names'].index(name_in_loaded)
+                id_in_loaded = state_dict['param_groups'][0]['params'][index_in_loaded_list]
+                # Copy the state of the corresponding parameter
+                if id_in_loaded in state_dict['state']:
+                    adapted_state_dict['state'][param_id] = clone_deepcopy(state_dict['state'][id_in_loaded])
+
+        return adapted_state_dict
+
+    optimizer2.register_load_state_dict_pre_hook(adapt_state_dict_ids)
+    optimizer2.load_state_dict(torch.load(PATH)) # The previous optimizer saved state_dict
+
+
+
+Instead of loading a state according to the order of parameters (the default approach),
+this hook can be used to load according to the parameters' names::
```

Comment:
```suggestion
As a third example, instead of loading a state according to the order of parameters (the default approach),
this hook can be used to load according to the parameters' names::
```